src/main/java/net/sf/jabref/gui/FindUnlinkedFilesDialog.java

```diff
@@ -495,6 +497,10 @@ private void startSearch() {
         if (Files.notExists(directory)) {
             directory = Paths.get(System.getProperty("user.dir"));
         }
+        if (!Files.isDirectory(directory)) {
```

Comment:
@simonharrer  The thing is about the Search Directory, not individual files
If accidentally a file (e.g.  xyz.pdf)  is selected or stored in the settings, the parent (e..g the directory) of this file is used.
And the FileChooser dialog only allows to select a directory, not a single file (not visible).
I tested all kind of cases and also when the folder is deleted etc.
That is also why I decided to implement the checks here, when the scan Button is clicked